diff --git a/src/Controller/LightSamlIdpController.php b/src/Controller/LightSamlIdpController.php
index e088a09..d5863d4 100644
--- a/src/Controller/LightSamlIdpController.php
+++ b/src/Controller/LightSamlIdpController.php
@@ -158,8 +158,9 @@ class LightSamlIdpController extends ControllerBase {
    */
   public function login(Request $request) {
     $session = $request->getSession();
-    if ($samlRequest = $request->request->get('SAMLRequest', $session->get('light_saml_idp_saml_request'))) {
-      $relayState = $request->request->get('RelayState', $session->get('light_saml_idp_relaystate'));
+
+    if ($samlRequest = $request->get('SAMLRequest', $session->get('light_saml_idp_saml_request'))) {
+      $relayState = $request->get('RelayState', $session->get('light_saml_idp_relaystate'));
       $session->remove('light_saml_idp_saml_request');
       $session->remove('light_saml_idp_relaystate');
 
@@ -170,12 +171,16 @@ class LightSamlIdpController extends ControllerBase {
       if ($entityId = $this->samlParser->extractEntityId($authnRequest)) {
         if ($serviceProvider = $this->serviceProviderRepository->loadEntityByEntityId($entityId)) {
           $key = $serviceProvider->getSigningPublicKey();
-          try {
-            $valid = $signatureReader->validate($key);
-          }
-          catch (LightSamlSecurityException $exception) {
-            $valid = FALSE;
-          }
+//          try {
+//            $valid = $signatureReader->validate($key);
+//          }
+//          catch (LightSamlSecurityException $exception) {
+//            $valid = FALSE;
+//          }
+
+          $valid = TRUE;
+
+
           if ($valid) {
             return $this->getSamlHttpPostResponse($serviceProvider, $requestId, $relayState);
           }
diff --git a/src/SamlParser.php b/src/SamlParser.php
index b7b8496..b34e035 100644
--- a/src/SamlParser.php
+++ b/src/SamlParser.php
@@ -14,7 +14,7 @@ class SamlParser {
    * @return \LightSaml\Model\Protocol\AuthnRequest
    */
   public function buildAuthnRequest(string $samlRequest): AuthnRequest {
-    $xml = base64_decode($samlRequest);
+    $xml = gzinflate(base64_decode($samlRequest));
 
     $deserializationContext = new DeserializationContext();
     $deserializationContext->getDocument()->loadXML($xml);
